Source/WarpX.cpp

```diff
@@ -151,6 +151,11 @@ WarpX::WarpX ()
 
     BackwardCompatibility();
 
+#ifdef WARPX_DIM_RZ
+    if (WarpX::maxwell_solver_id == MaxwellSolverAlgo::PSATD)
+        CheckGriddingForRZSpectral();
```

Comment:
Unfortunately, this doesn't work when it is put here. It needs to be called before the WarpX instance is created since that is where the max_grid and blocking_factor are read in and processed. A solution is to call this from main as before and have it read in the `algo.maxwell_solver` parameter and only do the changes if it is set to PSATD.

Reply:
Thank you!
I thought if I move this after `ReadParameters();` it might work, let me double-check and/or move it back to main.